### DISCUSSION TITLE
Feature Request: Import/Export from an Azure Repo or Github

### DISCUSSION THREAD

**Discussion by VexingCode** [Micke-K/IntuneManagement#285] [Ideas]
It would be awesome to have a feature allowing us to plug in Azure Repo or Github tokens/creds/etc., and backup or restore from the commits there.

**Comment by Micke-K** [comment]
Hello,

So sorry. I completely missed the discussions the last month. I've been too busy, so I haven't had much time looking into things.

Not sure how that would work. Do you know ho it gets an authentication token from any of them?

Cheers!

**Comment by VexingCode** [comment]
No problem at all! I definitely understand being slammed. It would be a "nice to have" sort of thing, and not required, but would be a huge feature enhancement IMO. For me, it would make the entire script a one stop shop and even easier to automate.

With Azure DevOps Repos and GitHub you can generate a Personal Access Token (or utilize SSH if you have that setup already) to interact.

https://learn.microsoft.com/en-us/azure/devops/repos/git/auth-overview?view=azure-devops
https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens

When I originally started this journey, I was looking at [Andrew Taylor's GitHub script](https://github.com/andrew-s-taylor/public/blob/main/Powershell%20Scripts/Intune/intune-backup-restore-withgui.ps1) from his [site](https://andrewstaylor.com/2022/12/07/intune-backing-up-and-restoring-your-environment-new-and-improved/), but then I found yours, which was amazing and has worked far better for me with virtually no fuss. His is able to do the upload to Azure or GitHub, etc., and It's the only feature that I miss.
